### DISCUSSION TITLE
NER Dates

### DISCUSSION THREAD

**Discussion by the1gofer** [explosion/spaCy#7833] [Help: Best practices]
Is their an easy way to filter no date dates from the results?  For example,  Im getting things like "78 years", when really I'm just trying to find specific dates.

**Comment by polm** [comment]
The pretrained models don't differentiate between those kinds of dates, so filtering out the ones you want needs to be a post-processing step. 

Not exactly sure what you want (is a date like "Aug 10" without a year valid? What about "July 4th"?), but a simple way to do this would be to feed the dates to a versatile date parser and discard anything that couldn't be parsed.

**Reply by the1gofer** [reply]
Do you have a suggestion for a library that might work?

**Reply by polm** [reply]
There's nothing I regularly use in Python, but it looks like [dateparser](https://github.com/scrapinghub/dateparser) has the necessary functionality. You'd need to disable its relative date parsing though.
